src/defs.hpp

```diff
@@ -87,6 +87,9 @@ struct RegionSize {
   int &nx(CoordinateDirection dir) { return nx_[dir - 1]; }
   const int &nx(CoordinateDirection dir) const { return nx_[dir - 1]; }
 
+  Real LogicalToActualPosition(Real u, CoordinateDirection dir) const {
```

Comment:
Can we get a one line comment what this function does, i.e., what the parameter `u` is (especially as it's a `Real` not  an `int`, which is what I'd have assumed from the function name)?

Reply:
added. This just moves from a block local coordinate system running from 0 to 1 to the global coordinate system.